odml/base.py

```diff
@@ -136,11 +136,17 @@ def __contains__(self, key):
             if (hasattr(obj, "name") and obj.name == key) or key == obj:
                 return True
 
-    def append(self, obj):
-        if obj.name in self:
-            raise KeyError("Object with the same name already exists! " + str(obj))
-        else:
-            super(SmartList, self).append(obj)
+    def append(self, *obj_tuple):
+        ll = len(obj_tuple)
+        for i in list(range(0,ll)):
+                if obj_tuple[i].name in self:
+                    raise KeyError("Object with the same name already exists! " + str(obj_tuple[i]))
+                from odml.section import BaseSection
+                from odml.doc import BaseDocument
```

Comment:
@jgrewe I agree, moved the imports outside the loop. AFAIU,  putting those imports into global scope within `base.py` creates circular dependency  as `odml.section` and `odml.doc` depends on `odml.base`. So imports stayed within the methods.  So to speak deferring the import until it is really needed.